internal/scaler/awsec2asg/aws_ec2_asg.go

```diff
@@ -82,7 +85,15 @@ func (s *Scaler) Type() scaler.Type {
 }
 
 func (s *Scaler) Register(ctx context.Context) error {
-	sess, err := session.NewSession()
+
+	var AwsRegion *string
+
+	if s.Region != nil {
+		AwsRegion = s.Region
+	}
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(*AwsRegion),
+	})
```

Comment:
Thanks for the update on this one :+1:

I notice that we are trying to make use of a nil pointer (`AwsRegion`) both when s.Region is set and not set. It it better to generally refrain from passing in `nil` values to library functions (as it might become a chance for libraries to get into error states)

So, I propose we go with an approach where we conditionally pass in the configuration to `NewSession` optionally based on the `s.Region` value like this

```suggestion
	var sessConfigs []*aws.Config
	if s.Region != nil {
		sessConfigs = append(sessConfigs, &aws.Config{
			Region: s.Region,
		})
	}
	sess, err := session.NewSession(sessConfigs...)
```

Reply:
@scriptnull when s.Region is nil, even in second case we would be passing an empty silce as an argument right?

Reply:
`session.NewSession` is a variadic function: https://gobyexample.com/variadic-functions - It can accept any number of arguments.

In the case of the second approach, we start with an empty array capable of holding `*aws.Config`  elements. If `s.Region` is nil, then this array will be empty and we are passing in the elements of the slice instead of the empty slice itself using the `...` operator to the `session.NewSession` function. Since our `sessConfigs` slice will be empty in this case, zero arguments will be sent to `session.NewSession` and it will behave as if no region is mentioned and try to get it from other places like the `AWS_REGION` env, aws config file, etc.